tests/unit/_experimental/dataset/test_dataset.py

```diff
@@ -52,6 +53,43 @@ def test__infer_datatype_value(uri: str, expected: str) -> None:
     assert _infer_datatype_value(uri) == expected
 
 
+def test__add_datatype() -> None:
+    df = pd.DataFrame(
+        dict(
+            locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
+        ),
+    )
+    add_datatype(df)
+    assert df[DATA_TYPE_FIELD].equals(
+        pd.Series([DatapointType.DOCUMENT, DatapointType.IMAGE, DatapointType.VIDEO, DatapointType.POINT_CLOUD]),
+    )
+
+
+def test__add_datatype__composite() -> None:
```

Comment:
in my understanding, composite supports arbitrary amounts of sub datapoints. should we test for 1 sub dp and more than 2 dps as well?

Reply:
Ya, your understanding is correct. I will add a test for 1 sub datapoint. Do you see benefit in having a test for 3 datapoints over 2?

Reply:
I'm going to add some edge cases which are missing from this test (eg. doubly nested composites)